com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GuiBuilder.java

```diff
@@ -405,7 +405,14 @@ public static SceneDesc.Builder transformScene(GuiBuilder builder, String input,
 
         if(builder != null) {
             // transform and register scene external resources (if compiling)
-            sceneBuilder.setScript(BuilderUtil.replaceExt(sceneBuilder.getScript(), ".gui_script", ".gui_scriptc"));
+            String scriptPath = sceneBuilder.getScript();
+            String suffix = BuilderUtil.getSuffix(scriptPath);
+            if (!suffix.isEmpty() && !suffix.equals("gui_script"))
```

Comment:
No need to check if it's empty too.

Reply:
no, we need to check if it's empty. Because not all the GUIs use scripts.